### PR TITLE
build: add and run synthtool

### PR DESCRIPTION
**Pull request by kolea2**
Towards #2382 

Note - the only handwritten files are `synth.py` and `.repo-metadata.json`. All other files are generated as a result of running `synthtool`.

### CODE REVIEW

File: .repo-metadata.json [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{
+  "name": "bigtable-hbase",
```
Comment (igorbernstein2) [inline .repo-metadata.json]: what is this name used for? Just pattern matching this looks different from the name in the native client:

In the native client:
repo name: java-bigtable
metadata name: google-cloud-bigtable

Here:
repo name: java-bigtable-hbase
metadata name: bigtable-hbase

File: .repo-metadata.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  "name_pretty": "Google Cloud Bigtable Hbase Client",
```
Comment (igorbernstein2) [inline .repo-metadata.json]: This also doesn't seem in line with the native client

Comment (kolea2) [reply .repo-metadata.json]: This is only used for README generation, which we are skipping

File: .repo-metadata.json (hunk 1 above)
Comment (kolea2) [reply .repo-metadata.json]: This was only used for generating the bug report template. https://github.com/googleapis/java-bigtable-hbase/pull/2412/commits/30c6aa041f70e56c5b9412aef03e4d0f7880b76b#diff-74c2649c999338ed7a2833d79d4b25c3R20. I've cleaned that up a little and added it to exclude from the generation.

**Pull request merged**